scripts/events_module/generate_events.py

```diff
@@ -949,88 +536,30 @@ def possible_lead_den_events(
                     continue
 
             elif "reputation" in event:
-                reputation = game.clan.reputation
-                # hostile
-                if (
-                    1 <= reputation <= 30
-                    and "hostile" not in event["reputation"]
-                    and "any" not in event["reputation"]
-                ):
-                    continue
-                # neutral
-                elif (
-                    31 <= reputation <= 70
-                    and "neutral" not in event["reputation"]
-                    and "any" not in event["reputation"]
-                ):
-                    continue
-                # welcoming
-                elif (
-                    71 <= reputation <= 100
-                    and "welcoming" not in event["reputation"]
-                    and "any" not in event["reputation"]
-                ):
+                if not event_for_reputation(event["reputation"]):
                     continue
 
             cat_info = event["m_c"]
-            if "status" in cat_info:
-                # special lost cat check
-                if event_type == "outsider":
-                    if cat.status not in [
-                        "loner",
-                        "rogue",
-                        "kittypet",
-                        "former Clancat",
-                        "exiled",
-                    ]:
-                        if "lost" not in cat_info["status"]:
-                            continue
-                    elif (
-                        cat.status.casefold() not in [x.casefold() for x in cat_info["status"]]
-                        and "any" not in cat_info["status"]
-                    ):
-                        continue
-                elif (
-                    cat.status not in cat_info["status"]
-                    and "any" not in cat_info["status"]
-                ):
-                    continue
-            if "age" in cat_info:
-                if cat.age not in cat_info["age"]:
-                    continue
-            if "trait" in cat_info:
-                if cat.personality.trait not in cat_info["trait"]:
-                    continue
-            if "skill" in cat_info:
-                has_skill = False
-                for _skill in cat_info["skill"]:
-                    split = _skill.split(",")
-
-                    if len(split) < 2:
-                        print("Cat skill incorrectly formatted", _skill)
-                        continue
-
-                    if cat.skills.meets_skill_requirement(split[0], int(split[1])):
-                        has_skill = True
-                        break
-                if not has_skill:
-                    continue
+            if not event_for_cat(
+                    cat_info=cat_info,
+                    cat=cat
+            ):
+                continue
 
             possible_events.append(event)
 
         return possible_events
 
 
 generate_events = GenerateEvents()
```

Comment:
May be able to delete this line, depends if it's used elsewhere.

Reply:
I'll have to do more finagling for that bit, since there's some leader den specific logic in here

Reply:
Sounds good, maybe make an enhancement post for it? Or just keep it in your mind palace lmao

Reply:
Yeah I'll make a post for it, else I shall def forget (and if i'm lucky, rusty or someone else will do it xD)